framework/src/play/templates/TemplateLoader.java

```diff
@@ -50,55 +48,55 @@ public static void cleanCompiledCache() {
      * @return The executable template
      */
     public static Template load(String path) {
-        Template template = null;
-        for (VirtualFile vf : Play.templatesPath) {
+        for (File vf : Play.templatesPath) {
             if (vf == null) {
                 continue;
             }
-            VirtualFile tf = vf.child(path);
+            File tf = new File(vf, path);
             boolean templateExists = tf.exists();
             if (!templateExists && Play.usePrecompiled) {
-                String name = tf.relativePath().replaceAll("\\{(.*)\\}", "from_$1").replace(":", "_").replace("..", "parent");
+                String name = Play.relativePath(tf).replaceAll("\\{(.*)\\}", "from_$1").replace(":", "_").replace("..", "parent");
                 templateExists = Play.getFile("precompiled/templates/" + name).exists();
             }
             if (templateExists) {
-                template = TemplateLoader.load(tf);
-                break;
+                return TemplateLoader.load(tf);
             }
         }
+
         /*
          * if (template == null) { //When using the old 'key = (file.relativePath().hashCode() + "").replace("-",
```

Comment:
comment totally useless, better remove it.

Reply:
This is out of scope of this PR, I would say. There is much more garbage to be removed, but let's do it in other PRs.